### PR TITLE
Fixed the X-Frame bug

### PR DESCRIPTION
**Pull request by blueelvis**
This is regarding this pull request.
 #10284

For some unknown reason, Github is not updating that PR with the commits so had to create a new pull request :(
If anyone has any idea what went wrong, please do let me know :)

### CODE REVIEW

**Comment by mattab** [comment]
> For some unknown reason, Github is not updating that PR with the commits so had to create a new pull request :( If anyone has any idea what went wrong

Maybe you didn't push your changes to Github / remote, and only pushed your changes to your local git repo?

File: core/Db/Adapter.php [hunk 1]
```diff
@@ -119,4 +119,23 @@ public static function getAdapters()
...
+     * @param string $adapterName
+     * @return bool
+     */
+    public static function isRecommendedAdapter($adapterName)
+    {
+        static $recommendedAdapters = array(
+            // Adapters recommended by Piwik. Needs to be in uppercase for string checking using in_array()
+            'PDO\MYSQL'
+        );
+        if(in_array($adapterName,$recommendedAdapters)) {
```
Comment (mattab) [inline core/Db/Adapter.php]: you can simply do `return in_array(...);`

File: core/View.php [hunk 2]
```diff
@@ -330,15 +330,17 @@ public function setContentType($contentType)
...
     public function setXFrameOptions($option = 'deny')
     {
         if ($option === 'deny' || $option === 'sameorigin') {
             $this->xFrameOptions = $option;
+            return;
         }
-        if ($option == 'allow') {
-            $this->xFrameOptions = null;
+        if (stripos($option, 'allow-from') !== false) {
+            $this->xFrameOptions = $option;
```
Comment (mattab) [inline core/View.php]: could you move this change to a separate PR? Ideally, a PR makes one fix only, and we use separate PRs for separate topics :+1:

**Comment by mattab** [comment]
Thanks for the PR, left a few comments.

**Comment by blueelvis** [comment]
@mattab - Separated the code. Pushed the changes. Please check :)

**Comment by tsteur** [comment]
@mattab the changes were made, can we merge this one?

File: core/View.php [hunk 3]
```diff
@@ -330,15 +330,17 @@ public function setContentType($contentType)
...
-     * @param string $option ('deny' or 'sameorigin')
+     * @param string $option ('deny' or 'sameorigin' or 'allow-from URI')
      */
     public function setXFrameOptions($option = 'deny')
     {
         if ($option === 'deny' || $option === 'sameorigin') {
             $this->xFrameOptions = $option;
+            return;
         }
-        if ($option == 'allow') {
```
Comment (mattab) [inline core/View.php]: @blueelvis I don't think  this code should be removed, since it's still used in https://github.com/piwik/piwik/blob/2.16.2/plugins/CoreAdminHome/OptOutManager.php#L203 and https://github.com/piwik/piwik/blob/master/plugins/Widgetize/Controller.php#L47-47

**Comment by mattab** [comment]
Hi @blueelvis 
Thank you for this proposed pull request.

Because it was last updated more than one month ago, it is our policy to close pull requests opened for a long time without updates. If you would like to continue work on the pull request, please simply ping us to have it re-opened (after you have pushed a new commit).

We hope you understand this and we look forward to seeing an update from you on this pull request or another one!

Thanks.
